### DISCUSSION TITLE
Delay when starting or stopping navigation

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#93] [Q&A] [answered]
There is a delay of more than one second when pressing the navigation button. Is this normal because of the many new features? It is not a big issue if this cannot be changed. It is just striking because before this took only milli-seconds.

**Comment by devemux86** [comment]
Thanks for the report.

Delay until the navigation panels appear on the screen or do you mean something else?
Is the GPS location already found and available on the screen before navigation starts?

What route, is it very long?

**Comment by devemux86** [comment]
Do you change the screen orientation (portrait / landscape) when you see this?

**Comment by DolomitesTom** [comment]
> Delay until the navigation panels appear on the screen 

yes
and good assumption: It only happens when using landscape mode. When staying in portrait mode, it works as before.

**Reply by DolomitesTom** [reply]
also on very short routes, so independent from the route length

**Reply by devemux86** [reply]
> It only happens when using landscape mode. When staying in portrait mode, it works as before.

I meant if you change the orientation while the navigation is already started?

I start navigation in both orientations with long or short routes and it starts immediately.

Can you export the app settings and send me the .pref file?

**Comment by DolomitesTom** [comment]
> I meant if you change the orientation while the navigation is already started?

pref file sent... no, when starting in portrait mode and setting is landscape mode (fixed): when pressing navigation button the delay occurs until landscape mode is set and the information bars appear. Then setting is portait (fixed), it starts directly.

**Reply by devemux86** [reply]
Thanks very much for the .pref file, I found what happens! :+1: 

> when starting in portrait mode and setting is landscape mode (fixed): when pressing navigation button the delay occurs until landscape mode is set and the information bars appear. Then setting is portait (fixed), it starts directly.

Currently if set a **fixed orientation** (portrait / landscape) in "Settings | Application | Display | Screen orientation"
and start navigation in a **different orientation**, the app restarts in the selected fixed orientation, this is the delay.

It did not happen in previous Cruiser versions, so I will check how this can be improved! :slightly_smiling_face:

**Reply by DolomitesTom** [reply]
It is always gratifying to see how well this works 🙂.

**Comment by DolomitesTom** [comment]
just recognized: when setting is SYSTEM and you just turn the display within an active navigation, then the navigation stops and it wants to import the last route again. I guess this is also connected to this mini-bug.

**Reply by devemux86** [reply]
> when setting is SYSTEM and you just turn the display within an active navigation, then the navigation stops and it wants to import the last route again

In this case navigation should automatically restart keeping the existing route?

> I guess this is also connected to this mini-bug

Correctly, it will work properly again in the next Cruiser update.

I will make another release soon, fixing the recent reports. :slightly_smiling_face:

**Reply by DolomitesTom** [reply]
> In this case navigation should automatically restart keeping the existing route?

yes please

**Comment by devemux86** [comment]
It will work properly again in the next Cruiser update (coming soon).

**Reply by DolomitesTom** [reply]
perfect thanks

**Comment by devemux86** [comment, marked as answer]
It should work properly again in [Cruiser 3.0.14](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

@DolomitesTom thanks for the report!
